### PR TITLE
python3Packages.pdoc: 14.1.0 -> 14.2.0

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

* packages pdoc checkPhase dep https://github.com/mitmproxy/pdoc-pyo3-sample-library
* pdoc diff: https://github.com/mitmproxy/pdoc/compare/v14.1.0...v14.2.0
* pdoc changelog: https://github.com/mitmproxy/pdoc/blob/v14.2.0/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 274521` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bx-py-utils</li>
    <li>python310Packages.bx-py-utils.dist</li>
    <li>python310Packages.jaraco-abode</li>
    <li>python310Packages.jaraco-abode.dist</li>
    <li>python310Packages.pdoc</li>
    <li>python310Packages.pdoc-pyo3-sample-library</li>
    <li>python310Packages.pdoc-pyo3-sample-library.dist</li>
    <li>python310Packages.pdoc.dist</li>
    <li>python311Packages.bx-py-utils</li>
    <li>python311Packages.bx-py-utils.dist</li>
    <li>python311Packages.jaraco-abode</li>
    <li>python311Packages.jaraco-abode.dist</li>
    <li>python311Packages.pdoc</li>
    <li>python311Packages.pdoc-pyo3-sample-library</li>
    <li>python311Packages.pdoc-pyo3-sample-library.dist</li>
    <li>python311Packages.pdoc.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, rustPlatform
+, cargo
+, rustc
+}:
+
+buildPythonPackage rec {
+  pname = "pdoc_pyo3_sample_library";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix:10]:
```suggestion
  pname = "pdoc-pyo3-sample-library";
```

File: pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  version = "1.0.11";
+  format = "pyproject";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix:12]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/pdoc/default.nix [hunk 3]
```diff
@@ -13,7 +14,7 @@
 
 buildPythonPackage rec {
   pname = "pdoc";
-  version = "14.1.0";
+  version = "14.2.0";
   disabled = pythonOlder "3.8";
 
   format = "pyproject";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pdoc/default.nix:20]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
```
Comment (dotlambda) [inline pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix:15]:
```suggestion
    pname = "pdoc_pyo3_sample_library";
    inherit version;
```

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 274521` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bx-py-utils</li>
    <li>python310Packages.bx-py-utils.dist</li>
    <li>python310Packages.jaraco-abode</li>
    <li>python310Packages.jaraco-abode.dist</li>
    <li>python310Packages.pdoc</li>
    <li>python310Packages.pdoc-pyo3-sample-library</li>
    <li>python310Packages.pdoc-pyo3-sample-library.dist</li>
    <li>python310Packages.pdoc.dist</li>
    <li>python311Packages.bx-py-utils</li>
    <li>python311Packages.bx-py-utils.dist</li>
    <li>python311Packages.jaraco-abode</li>
    <li>python311Packages.jaraco-abode.dist</li>
    <li>python311Packages.pdoc</li>
    <li>python311Packages.pdoc-pyo3-sample-library</li>
    <li>python311Packages.pdoc-pyo3-sample-library.dist</li>
    <li>python311Packages.pdoc.dist</li>
  </ul>
</details>

**Comment by pbsds** [comment]
@ofborg python310Packages.pdoc python311Packages.pdoc python310Packages.pdoc.passthru.tests python311Packages.pdoc.passthru.tests

**Comment by felixalbrigtsen** [comment]
Result of `nixpkgs-review pr 274521` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.pdoc-pyo3-sample-library</li>
    <li>python310Packages.pdoc-pyo3-sample-library.dist</li>
    <li>python311Packages.pdoc-pyo3-sample-library</li>
    <li>python311Packages.pdoc-pyo3-sample-library.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python310Packages.bx-py-utils</li>
    <li>python310Packages.bx-py-utils.dist</li>
    <li>python310Packages.jaraco-abode</li>
    <li>python310Packages.jaraco-abode.dist</li>
    <li>python310Packages.pdoc</li>
    <li>python310Packages.pdoc.dist</li>
    <li>python311Packages.bx-py-utils</li>
    <li>python311Packages.bx-py-utils.dist</li>
    <li>python311Packages.jaraco-abode</li>
    <li>python311Packages.jaraco-abode.dist</li>
    <li>python311Packages.pdoc</li>
    <li>python311Packages.pdoc.dist</li>
  </ul>
</details>

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 274521` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bx-py-utils</li>
    <li>python310Packages.bx-py-utils.dist</li>
    <li>python310Packages.jaraco-abode</li>
    <li>python310Packages.jaraco-abode.dist</li>
    <li>python310Packages.pdoc</li>
    <li>python310Packages.pdoc-pyo3-sample-library</li>
    <li>python310Packages.pdoc-pyo3-sample-library.dist</li>
    <li>python310Packages.pdoc.dist</li>
    <li>python311Packages.bx-py-utils</li>
    <li>python311Packages.bx-py-utils.dist</li>
    <li>python311Packages.jaraco-abode</li>
    <li>python311Packages.jaraco-abode.dist</li>
    <li>python311Packages.pdoc</li>
    <li>python311Packages.pdoc-pyo3-sample-library</li>
    <li>python311Packages.pdoc-pyo3-sample-library.dist</li>
    <li>python311Packages.pdoc.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix [hunk 5]
```diff
@@ -39,5 +40,6 @@ buildPythonPackage rec {
     homepage = "https://github.com/mitmproxy/pdoc-pyo3-sample-library";
     license = lib.licenses.mit;
     maintainers = [ lib.maintainers.pbsds ];
+    broken = stdenv.isDarwin; # -liconv
```
Comment (dotlambda) [inline pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix:43]: Instead add `libiconv` to `buildInputs` on Darwin.

Comment (pbsds) [reply pkgs/development/python-modules/pdoc-pyo3-sample-library/default.nix:43]: Huh, TIL about `darwin.libiconv`

**Pull request merged**